lightning/src/ln/functional_tests.rs

```diff
@@ -8503,3 +8503,119 @@ fn test_htlc_no_detection() {
         connect_blocks(&nodes[0], ANTI_REORG_DELAY - 1, 201, true, header_201.block_hash());
         expect_payment_failed!(nodes[0], our_payment_hash, true);
 }
+
+#[test]
+fn test_unordered_onchain_htlc_settlement() {
```

Comment:
As a general point, I don't know about other contributors opinion, but I'm leaning towards splitting test coverage in their own commit.

Reply:
for people that practice test-driven-development, that would be somewhat unnatural.  it also seems to go against commits being a unit of work.

Reply:
Leaving as is for now, let me know if you feel strongly @ariard? I tend to agree that it seems more conventional to combine test changes into the code changes they're testing.

Reply:
Okay I don't feel strongly about it, even if I consider it's easing review. At least, this is the opinion on Core-side.